modules/standard/CTypes.chpl

```diff
@@ -548,13 +553,29 @@ module CTypes {
   @chpldoc.nodoc
   inline operator :(const ref x:c_array, type t:c_ptrConst(?e))
       where x.eltType == e {
-    return c_ptrTo(x[0]):c_ptrConst(e);
+    return c_ptrToConst(x[0]);
   }
   @chpldoc.nodoc
   inline operator :(ref x:c_array, type t:c_ptr(void)) {
     return c_ptrTo(x[0]):c_ptr(void);
   }
   @chpldoc.nodoc
+  inline operator :(x:c_ptr, type t:c_ptr(void)) {
+    return __primitive("cast", t, x);
+  }
+  @chpldoc.nodoc
+  inline operator :(x:c_ptrConst, type t:c_ptr(void)) {
+    return __primitive("cast", t, x);
+  }
+  @chpldoc.nodoc
+  inline operator :(x:c_ptr(void), type t:c_ptr) {
+    return __primitive("cast", t, x);
+  }
+  @chpldoc.nodoc
+  inline operator :(x:c_ptr(void), type t:c_ptrConst) {
```

Comment:
Just curious, do we have `c_ptrConst(void)` as well? If we do, will need some related overloads.

Reply:
I think it's possible based on this test: `test/types/cptr/const/c_ptrConst_void.chpl`

I've added several new casts